src/shipit_static_analysis/shipit_static_analysis/report/mozreview.py

```diff
@@ -153,6 +158,58 @@ def __init__(self, api_root, review_request_id, diff_revision,
         self._destfile_to_file = {}
         self._file_to_diffdata = {}
 
+        # Load current user id
+        try:
+            self.user = self.api_root.get_session().get_user()
+        except Exception as e:
+            logger.error('Failed to retrieve current MozReview session user')
+            raise
+
+        # Load all existing issues
+        self.existing_comments = self.list_comments()
+        logger.info('Found {} existing comments on review'.format(len(self.existing_comments)))
+
+    def list_comments(self):
+        '''
+        List all existing comments on current Review
+        '''
+        issues = []
+        cache = {}
+
+        def from_cache(comment, name, loader):
+            # Cache sub parts ids to avoid too many slow requests
+            link = comment._payload['links'][name]['href']
+            value = cache.get(link)
+            if not value:
+                # Populate cache
+                value = cache[link] = loader()
+                logger.debug('Loaded mozreview extra data', comment=comment['id'], link=link)
```

Comment:
Nit: I find the comment and debug message aren't very clear: what is being cached here? (What are "sub parts ids" or "mozreview extra data" exactly?)

Reply:
This is a common boilerplate code used to load a Mozreview user and filediff (see lines just below).
The mozreview API forces us to load related objects as separate resource, so this code is just here to avoid repeating identical requests and speed up the process

Reply:
Thanks! But what I meant is that the comment / debug message aren't very explicit, e.g. to someone unfamiliar with MozReview who might need to debug or refactor this code one day.

Maybe add a comment like `We cache details about MozReview comments (by calling API sub-resources like 'user', 'filediff', etc.)`?

Reply:
Ok i see your point, i added an explicit comment.

Reply:
Perfect, thanks!